src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -0,0 +1,531 @@
+            "ExtendedLocation",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ExtendedLocationTypes = self.__cmd.get_models(
+            "ExtendedLocationTypes",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+
+
+class AKSCreateContext:
+    # Used to store intermediate variables (usually this stores the dynamically completed value of the parameter,
+    # which has not been decorated into the `mc` object, and some pure intermediate variables (such as the
+    # subscription ID)) and a copy of the original function parameters, and provide "getter" methods for all
+    # parameters.
+    # To dynamically complete a parameter or check the validity of a parameter, please provide a "getter" function
+    # named `get_xxx`, where `xxx` is the parameter name. In this function, the process of obtaining parameter
+    # values, dynamic completion (optional), and validation (optional) should be followed. The obtaining of
+    # parameter values should further follow the order of obtaining from the `mc` object, from the intermediates,
+    # or from the original value.
+    # Attention: In case of checking the validity of parameters, be sure not to set the `enable_validation` to
+    # `True` to avoid loop calls, when using the getter function to obtain the value of other parameters.
+    # Attension: After the parameter is dynamically completed, it must be added to the intermediates; and after
+    # the parameter is decorated into the `mc` object, the corresponding intermediate should be deleted.
+    def __init__(self, cmd: AzCliCommand, raw_parameters: Dict):
+        self.cmd = cmd
+        self.raw_param = raw_parameters
+        self.intermediates = dict()
+        self.mc = None
+
+    def attach_mc(self, mc):
+        if self.mc is None:
+            self.mc = mc
+        else:
+            msg = "the same" if self.mc == mc else "different"
+            raise CLIInternalError(
+                "Attempting to attach the `mc` object again, the two objects are {}.".format(
+                    msg
+                )
+            )
+
+    def get_intermediate(self, variable_name: str, default_value: Any = None):
+        if variable_name not in self.intermediates:
+            msg = "The intermediate '{}' does not exist, return default value '{}'.".format(
+                variable_name, default_value
+            )
+            logger.debug(msg)
+        return self.intermediates.get(variable_name, default_value)
+
+    def set_intermediate(
+        self, variable_name: str, value: Any, overwrite_exists: bool = False
+    ):
+        if variable_name in self.intermediates:
+            if overwrite_exists:
+                msg = "The intermediate '{}' is overwritten. Original value: '{}', new value: '{}'.".format(
+                    variable_name, self.intermediates.get(variable_name), value
```

Comment:
same here.

Reply:
No worry, the outer if statement has checked the existence of the object, even if the value is None, it can be automatically converted to a string for output.